src/extension.ts

```diff
@@ -1731,9 +1736,34 @@ async function deleteContextKubernetes(explorerNode: explorer.KubernetesObject)
     }
 }
 
-async function useNamespaceKubernetes(explorerNode: explorer.KubernetesObject) {
-    if (await kubectlUtils.switchNamespace(kubectl, explorerNode.id)) {
-        refreshExplorer();
+async function useNamespaceKubernetes(explorerNode?: explorer.KubernetesObject) {
+    if (explorerNode) {
+        if (await kubectlUtils.switchNamespace(kubectl, explorerNode.id)) {
+            refreshExplorer();
+        }
+    } else {
+        promptKindName([kuberesources.allKinds.namespace], 'get', undefined, (resource) => {
+            if (resource) {
+                const currentNS = kubectlUtils.currentNamespace(kubectl);
+                currentNS.then((ns) => {
+                    //resource will be of format <kind>/<name>, hence get the substring
+                    let toSwitchNamespace = '' + resource;
+                    toSwitchNamespace = toSwitchNamespace.substring(toSwitchNamespace.lastIndexOf('/') + 1);
+                    //Switch if an only if the currentNS and toSwitchNamespace are different
+                    if (toSwitchNamespace && ns !== toSwitchNamespace) {
+                        const promiseSwitchNS = kubectlUtils.switchNamespace(kubectl, toSwitchNamespace);
+                        promiseSwitchNS.then((data) => {
+                            if (data) {
+                                refreshExplorer();
+                            }
+                        });
+                    }
+                }).catch((err) => {
+                    host.showErrorMessage(err);
+                });
+
+            }
+        }, 'Type the namespace name or press enter to select from available');
```

Comment:
Per previous comment... this placeholder is 20 lines of multi-indent-level code away from the function it's used in... definitely needs to move earlier in the args list or into the opts record.